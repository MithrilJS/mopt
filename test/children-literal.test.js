"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("literal children", () => {
        it("should support single literal children (string)", () =>
            expect(
                code(`m("1", "2")`)
            )
            .toMatchSnapshot()
        );

        it("should support single literal children (number)", () =>
            expect(
                code(`m("1", 2)`)
            )
            .toMatchSnapshot()
        );

        it("should support single literal children (boolean)", () =>
            expect(
                code(`m("1", true)`)
            )
            .toMatchSnapshot()
        );

        it("should support single literal children (template)", () =>
            expect(
                code(`m("1", \`2\`)`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + single literal children children", () =>
            expect(
                code(`m("1", { title : "bar" }, "2")`)
            )
            .toMatchSnapshot()
        );

        it("should support multiple literal children", () =>
            expect(
                code(`m("1", "2", "3")`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + multiple children", () =>
            expect(
                code(`m("1", { title : "bar" }, "2", "3")`)
            )
            .toMatchSnapshot()
        );

        it("should not transform invocations containing identifiers", () =>
            // Identifiers can't be resolved at compile time, so ignore
            expect(
                code(`m(".fooga", identifier)`)
            )
            .toMatchSnapshot()
        );
    });
});
