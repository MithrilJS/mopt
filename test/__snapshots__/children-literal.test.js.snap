// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children literal children should not transform invocations containing identifiers 1`] = `"m(\\".fooga\\",identifier);"`;

exports[`Children literal children should support attrs + multiple children 1`] = `"m.vnode(\\"1\\",undefined,{title:\\"bar\\"},[m.vnode(\\"#\\",undefined,undefined,\\"2\\",undefined,undefined),m.vnode(\\"#\\",undefined,undefined,\\"3\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children literal children should support attrs + single literal children children 1`] = `"m.vnode(\\"1\\",undefined,{title:\\"bar\\"},undefined,\\"2\\",undefined);"`;

exports[`Children literal children should support multiple literal children 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"#\\",undefined,undefined,\\"2\\",undefined,undefined),m.vnode(\\"#\\",undefined,undefined,\\"3\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children literal children should support single literal children (boolean) 1`] = `"m.vnode(\\"1\\",undefined,undefined,undefined,true,undefined);"`;

exports[`Children literal children should support single literal children (number) 1`] = `"m.vnode(\\"1\\",undefined,undefined,undefined,2,undefined);"`;

exports[`Children literal children should support single literal children (string) 1`] = `"m.vnode(\\"1\\",undefined,undefined,undefined,\\"2\\",undefined);"`;

exports[`Children literal children should support single literal children (template) 1`] = `"m.vnode(\\"1\\",undefined,undefined,undefined,\`2\`,undefined);"`;
